Makefile

```diff
@@ -6,14 +6,29 @@ export HEAPAPPID := 892134159
 staticcheck := /home/runner/go/bin/staticcheck
 
 ifeq ($(OS),Windows_NT)     # is Windows_NT on XP, 2000, 7, Vista, 10...
-    detected_OS := Windows
+	detected_OS := Windows
 else
-    detected_OS := $(shell uname)  # same as "uname -s"
+	detected_OS := $(shell uname)  # same as "uname -s"
 endif
 
+HAS_STATICCHECK := $(shell command -v staticcheck 2> /dev/null)
+HAS_ERRCHECK := $(shell command -v errcheck 2> /dev/null)
+GO_VERSION := $(shell go version | cut -d' ' -f3)
 
 all: test
 
+check_environment:
+ifneq ("${GO_VERSION}", "go1.16.6")
+	$(error "requires go version 1.16.6 or higher")
+endif
+ifndef HAS_STATICCHECK
+	$(error "staticcheck is not on PATH")
```

Comment:
Perhaps we should declare required tools in `go.mod`, e.g.:
https://marcofranssen.nl/manage-go-tools-via-go-modules#tldr